### DISCUSSION TITLE
Compress-Archive -Update doesn't update the file?

### DISCUSSION THREAD

**Discussion by tboyle-gresham** [PowerShell/PowerShell#18651] [Q&A] [answered]
Hi, I'm trying to create a simple script to update a file within a zip, but running into some issues.
MyZip.zip contains some subdirectories, and the file I want to update (myFile.txt) is inside dir1/dir2/dir3:

![image](https://user-images.githubusercontent.com/42770456/203504857-e1dd613a-7313-4c43-b2f6-7cabb314f06b.png)

I'm then running the following command to try and update this file with a new version of myFile.txt (which is nested inside dir1/dir2/dir3 on my file system so that it gets updated into the right directory in the zip):

`Compress-Archive -Path .\dir1 -Update -DestinationPath .\myZip.zip`

But this is resulting in the new version of the file being added to the directory, but not replacing the old version:

![image](https://user-images.githubusercontent.com/42770456/203505560-646fa94d-bab0-48a0-92ac-53733e0fbeca.png)

What doesn't make sense after this, is that when I then go and update the source file, and rerun the same command, the new version of the file is updated correctly, and it doesn't make a third one?

![image](https://user-images.githubusercontent.com/42770456/203508181-072801f4-3400-4e9f-9c9a-ce86ce25aaa0.png)

Why wouldn't this work for the original file in the zip?
Am I doing something wrong, or do I need to change how I'm doing this?

Thanks in advance.

**Comment by lpatalas** [comment, marked as answer]
@tboyle3 I did what you described and it works fine on my machine. If you weren't able to solve it yet then could you attach that zip file here and paste the output of `$PSVersionTable` to see what OS/version you're using?

**Reply by tboyle-gresham** [reply]
@lpatalas I have been able to find a workaround for this, which included using the 7zip CLI to recursively search and delete the file first, then use Compress-Archive to add the new file in:

`./7z.exe d 'C:\Users\tboyle\Downloads\myZip.zip' 'myFile.txt' -r`
`Compress-Archive -Path .\dir1 -Update -DestinationPath .\myZip.zip`

$PSVersionTable output below:

PS C:\Users\tboyle> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.1682
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1682
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

**Reply by tboyle-gresham** [reply]
[myFile.txt](https://github.com/PowerShell/PowerShell/files/10111150/myFile.txt)
[myZip.zip](https://github.com/PowerShell/PowerShell/files/10111152/myZip.zip)

@lpatalas files attached if you fancy trying with these myZip.zip contains the "original" file, and the attached myFile.txt is the "updated" file.
The myFile.txt exists in `C:\users\tboyle\Downloads\dir1\dir2\dir3\myFile.txt`

**Reply by lpatalas** [reply]
You're using PowerShell 5.1 aka the Desktop Edition. That was the missing part of the puzzle ;) I can also reproduce it on that version but it works correctly in PowerShell 7.3. Could you switch to the new version instead?

**Reply by tboyle-gresham** [reply]
@lpatalas thank you! Updated and this works now! Really annoying that desktop edition is only 5.1 it seems!
